GraphQL integration errors on multiple use of enum type

The current GraphQL integration produces an error when an enum is used multiple times in a GraphQL model:

```
Caused by: java.lang.IllegalArgumentException: Enum type with name 'PaymentMethod' is registered multiple times: org.acme.business.invoice.entity.PaymentMethod
	at com.blazebit.persistence.integration.graphql.GraphQLEntityViewSupportFactory.getScalarType(GraphQLEntityViewSupportFactory.java:394)
	at com.blazebit.persistence.integration.graphql.GraphQLEntityViewSupportFactory.getElementType(GraphQLEntityViewSupportFactory.java:343)
	at com.blazebit.persistence.integration.graphql.GraphQLEntityViewSupportFactory.create(GraphQLEntityViewSupportFactory.java:193)
	at org.acme.GraphQLProvider.buildSchema(GraphQLProvider.java:56)
	at org.acme.GraphQLProvider.init(GraphQLProvider.java:46)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,6 +25,7 @@ None yet
 * Fix set operand wrapping for PostgreSQL
 * Prefer abstract class method over interface method in metamodel if attribute mappings are equal
 * Validate that entity type for creatable entity view is non-abstract
+* Fix GraphQL integration issue when encountering enum types multiple times
 
 ### Backwards-incompatible changes
 

--- a/integration/graphql/src/main/java/com/blazebit/persistence/integration/graphql/GraphQLEntityViewSupportFactory.java
+++ b/integration/graphql/src/main/java/com/blazebit/persistence/integration/graphql/GraphQLEntityViewSupportFactory.java
@@ -390,9 +390,7 @@ public class GraphQLEntityViewSupportFactory {
         if (typeName == null) {
             if (javaType.isEnum()) {
                 typeName = getTypeName(javaType);
-                if (typeRegistry.getType(typeName).isPresent()) {
-                    throw new IllegalArgumentException("Enum type with name '" + typeName + "' is registered multiple times: " + javaType.getName());
-                } else {
+                if (!typeRegistry.getType(typeName).isPresent()) {
                     List<EnumValueDefinition> enumValueDefinitions = new ArrayList<>();
                     for (Enum<?> enumConstant : (Enum<?>[]) javaType.getEnumConstants()) {
                         enumValueDefinitions.add(new EnumValueDefinition(enumConstant.name(), new ArrayList<>(0)));

